RawSpeed/DngDecoderSlices.cpp

```diff
@@ -143,6 +143,204 @@ my_error_throw (j_common_ptr cinfo)
   ThrowRDE("JPEG decoder error!");
 } 
 
+#ifdef HAVE_ZLIB
+// decodeFPDeltaRow(): MIT License, copyright 2014 Javier Celaya <jcelaya@gmail.com>
+static void decodeFPDeltaRow(unsigned char * src, unsigned char * dst, size_t tileWidth, size_t realTileWidth, unsigned int bytesps, int factor)
+{
+  // DecodeDeltaBytes
+  for (size_t col = factor; col < realTileWidth*bytesps; ++col) {
+    src[col] += src[col - factor];
+  }
+  // Reorder bytes into the image
+  // 16 and 32-bit versions depend on local architecture, 24-bit does not
+  if (bytesps == 3) {
+    for (size_t col = 0; col < tileWidth; ++col) {
+      dst[col*3] = src[col];
+      dst[col*3 + 1] = src[col + realTileWidth];
+      dst[col*3 + 2] = src[col + realTileWidth*2];
+    }
+  } else {
+    union X { unsigned int x; unsigned char c; };
+    if (((union X){1}).c) {
+    for (size_t col = 0; col < tileWidth; ++col) {
+      for (size_t byte = 0; byte < bytesps; ++byte)
+        dst[col*bytesps + byte] = src[col + realTileWidth*(bytesps-byte-1)];  // Little endian
+    }
```

Comment:
And since this is just one commit, also *please* *try* to [format](https://github.com/darktable-org/rawspeed/blob/develop/.clang-format) your new functions properly.
Only these few new functions, not the whole file/etc.